utils/youtube.js

```diff
@@ -0,0 +1,51 @@
+const stringSimilarity = require('string-similarity');
+const cheerio = require('cheerio');
+
+const { asyncRequest } = require('./utils.js');
+
+const baseUrl = 'https://www.youtube.com/results?search_query=';
+
+const MATCH_SENSITIVITY = 0.2;
+const INVALID_TITLE_WORDS = /\blyrics\b|\bmirrored\b|\bteaser\b|\btrailer\b|\bdance practice\b/;
+const MV_SCOPING_WORDS = /\bmv\b|\bm\/v\b|\bmusic video\b/;
+
+function _cleanSearchString(str) {
+    return str.toLowerCase().replace(' by ', ' ').replace(' - ', '').replace(/\s\s+/g, ' ').trim();
```

Comment:
Is this because `by` and `-` give worse search results for some reason?

Reply:
I think titles tend to not contain these words, but they're more colloquial for searching, so I filtered them out to make people's searches more generalized. I could consolidate it into a single regex though.